packages/lodestar/src/network/gossip/scoringParameters.ts

```diff
@@ -0,0 +1,275 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import {allForks} from "@chainsafe/lodestar-beacon-state-transition";
+import {IChainForkConfig} from "@chainsafe/lodestar-config";
+import {ATTESTATION_SUBNET_COUNT, SLOTS_PER_EPOCH, TARGET_AGGREGATORS_PER_COMMITTEE} from "@chainsafe/lodestar-params";
+import {Context, ILogger} from "@chainsafe/lodestar-utils";
+import {PeerScoreThresholds} from "libp2p-gossipsub/src/score";
+import {defaultTopicScoreParams, PeerScoreParams, TopicScoreParams} from "libp2p-gossipsub/src/score/peer-score-params";
+import {Eth2Context} from "../../chain";
+import {IForkDigestContext} from "../../util/forkDigestContext";
+import {FORK_EPOCH_LOOKAHEAD, getCurrentAndNextFork} from "../forks";
+import {IGossipsubModules} from "./gossipsub";
+import {GossipType} from "./interface";
+import {stringifyGossipTopic} from "./topic";
+
+export const GOSSIP_D = 8;
+export const GOSSIP_D_LOW = 6;
+export const GOSSIP_D_HIGH = 12;
+
+const MAX_IN_MESH_SCORE = 10.0;
+const MAX_FIRST_MESSAGE_DELIVERIES_SCORE = 40.0;
+const BEACON_BLOCK_WEIGHT = 0.5;
+const BEACON_AGGREGATE_PROOF_WEIGHT = 0.5;
+const VOLUNTARY_EXIT_WEIGHT = 0.05;
+const PROPOSER_SLASHING_WEIGHT = 0.05;
+const ATTESTER_SLASHING_WEIGHT = 0.05;
+
+const beaconAttestationSubnetWeight = 1 / ATTESTATION_SUBNET_COUNT;
+const maxPositiveScore =
+  (MAX_IN_MESH_SCORE + MAX_FIRST_MESSAGE_DELIVERIES_SCORE) *
+  (BEACON_BLOCK_WEIGHT +
+    +BEACON_AGGREGATE_PROOF_WEIGHT +
+    beaconAttestationSubnetWeight * ATTESTATION_SUBNET_COUNT +
+    VOLUNTARY_EXIT_WEIGHT +
+    PROPOSER_SLASHING_WEIGHT +
+    ATTESTER_SLASHING_WEIGHT);
+
+/**
+ * The following params is implemented by Lighthouse at
+ * https://github.com/sigp/lighthouse/blob/b0ac3464ca5fb1e9d75060b56c83bfaf990a3d25/beacon_node/eth2_libp2p/src/behaviour/gossipsub_scoring_parameters.rs#L83
+ */
+export const gossipScoreThresholds: PeerScoreThresholds = {
+  gossipThreshold: -4000,
+  publishThreshold: -8000,
+  graylistThreshold: -16000,
+  acceptPXThreshold: 100,
+  opportunisticGraftThreshold: 5,
+};
+
+type MeshMessageInfo = {
+  decaySlots: number;
+  capFactor: number;
+  activationWindow: number;
+  currentSlot: number;
+};
+
+export class GossipPeerScoreParamsBuilder {
```

Comment:
This should not be a class, it holds not state and is only called one. Please convert to function that returns the params